### PR TITLE
Add documentation for gradle scripts

### LINKED ISSUE #591: Add documentation for ODK 2 Gradle scripts
The ODK 2 build system and gradle scripts are not set up in the typical way and can be confusing to use or change. This should be documented for developers.

### PR DESCRIPTION
**Pull request by jbeorse**
closes #591 

## What is included in this PR?
Adds a brief overview of the build system for the ODK 2 Android tools, and the Gradle Config repository.

### CODE REVIEW

File: odk2-src/build-scripts.rst [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+  /opendatakit/
+      /androidlibrary/
+      /androidcommon/
+      /app-designer/
+      /gradle-config/
+      /scan/
+      /sensorsframework/
+      /sensorsinterface/
+      /services/
+      /suitcase/
```
Comment (wbrunette) [inline odk2-src/build-scripts.rst]: suitcase is not an android tool

File: odk2-src/build-scripts.rst [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+------------------------
+
+The build scripts for the Android tools expect a particular directory structure. They expect a parent directory that contains each of them at the same level. This is **optional**.
+
+If you had all of the Android tools checked out your directory structure would look like this::
+
+  /opendatakit/
+      /androidlibrary/
+      /androidcommon/
+      /app-designer/
```
Comment (wbrunette) [inline odk2-src/build-scripts.rst]: app-designer is not an android tool

File: odk2-src/build-scripts.rst [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+      /gradle-config/
+      /scan/
+      /sensorsframework/
+      /sensorsinterface/
+      /services/
+      /suitcase/
+      /survey/
+      /sync-client/
```
Comment (wbrunette) [inline odk2-src/build-scripts.rst]: sync-client is not an android tool

File: odk2-src/build-scripts.rst [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    - If the library projects are not present in the above configuration then a prebuilt binary will be downloaded according to the flavor you are building. For example, new binaries are posted on Snapshot for each commit, or on Master for each release.
+
+  - **Gradle Config**: If the gradle-config project is present in the above configuration, the gradle files in that folder will be used. Otherwise the release version specifies in :file:`settings.gradle` will be used.
+
+.. _build-scripts-building:
+
+
+Building the Android Tools
+-------------------------------
+
```
Comment (wbrunette) [inline odk2-src/build-scripts.rst]: Add a discussion about that we have two flavor dimensions. Describe each dimension's purpose and possible values.

File: odk2-src/build-scripts.rst [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+Using the Snapshot flavor assumes you are running in the *development* branch. If you are using the *master* branch, use the *master* flavor instead:
+
+.. code-block:: console
+
+  ./gradlew clean assembleMasterBasic
+
+.. warning::
+
+  The ODK 2 tools do not accept pull requests to master. If you are contributing, please use the development branch.
```
Comment (wbrunette) [inline odk2-src/build-scripts.rst]: I would state it differently, that ODK 2 prefers all pull requests to master. In an unusual circumstances when development is undergoing heavy change we may accept pull requests to 'demo' or 'master' depending on the level of incompatibility that might exist.

File: odk2-src/build-scripts.rst [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This file contains miscellaneous Gradle tasks necessary to the ODK 2 tools. Mostly these exist to make Jenkins or Artifactory work.
+
+.. _build-scripts-external-uitests:
+
+:file:`uitests.gradle`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This file contains tasks to make the UI tests work on a headless build server. In particular they disable animations and grant external storage permissions.
```
Comment (wbrunette) [inline odk2-src/build-scripts.rst]: remove headless :)

File: odk2-src/build-scripts.rst [hunk 7]
```diff
@@ -58,24 +55,41 @@ There are two cases where this directory structure makes a difference:
...
 
-  The ODK 2 tools do not accept pull requests to master. If you are contributing, please use the development branch.
+  The ODK 2 tools prefers pull requests to *development*. In unusual circumstances when *development* is undergoing heavy change we may accept pull requests to *demo* or *master* depending on the level of incompatibility that might exist.
+
+The other dimension determines whether to apply changes necessary to run the UI tests. The two options are:
+
+  - **Basic** is used for normal builds
+  - **Uitest** is used for builds that will run the UI tests.
+
+Therefor, if you wanted to build the normal version of the *master* branch, you would run:
```
Comment (linl33) [inline odk2-src/build-scripts.rst]: Therefore

File: odk2-src/build-scripts.rst [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+
+  - **Gradle Config**: If the gradle-config project is present in the above configuration, the gradle files in that folder will be used. Otherwise the release version specifies in :file:`settings.gradle` will be used.
+
+.. _build-scripts-building:
+
+
+Building the Android Tools
+-------------------------------
+
+The simplest way to build the tools is often to press the build button in Android Studio. However, the command line can also be used. To invoke the gradle wrapper, enter the root level of the project to be built and run a command that looks like this:
```
Comment (linl33) [inline odk2-src/build-scripts.rst]: We should add note here that you need to have the correct build variant set in Android Studio. This is important when you don't have androidlibrary and/or androidcommon checked out.

File: odk2-src/build-scripts.rst [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+------------------------
+
+These build files are centralized in the `Gradle Config repository <https://github.com/opendatakit/gradle-config>`_. They included shared configuration, versions, and tasks.
+
+.. _build-scripts-external-variables:
+
+:file:`variables.gradle`
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This file contains all the versions and variables strings shared among the projects. Most notably this includes the release code version, the compile targets, the :program:`Java` version, and the composed project build and flavor names.
```
Comment (linl33) [inline odk2-src/build-scripts.rst]: This file stores the variant names not the flavor names. The flavor names are in `publish.gradle`.

**Pull request merged**